# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1424201585 18000
#      Tue Feb 17 14:33:05 2015 -0500
# Node ID 08033d58343501770f7cf188daed52530547d726
# Parent  bf25e9f7c22ad4695c7a363a869cf626db59d72c
more consistently use _CPPFLAGS and _LDFLAGS variables

* common.mk, configure.ac, toplev.cc, libinterp/dldfcn/module-files,
oct-conf.in.h: Use FLTK_CPPFLAGS, FLTK_LDFLAGS, FONTCONFIG_CPPFLAGS,
FT2_CPPFLAGS for consistency with other library compiler flag variables.

diff --git a/build-aux/common.mk b/build-aux/common.mk
--- a/build-aux/common.mk
+++ b/build-aux/common.mk
@@ -220,17 +220,18 @@ FFTW3_LIBS = @FFTW3_LIBS@
 FFTW3F_CPPFLAGS = @FFTW3F_CPPFLAGS@
 FFTW3F_LDFLAGS = @FFTW3F_LDFLAGS@
 FFTW3F_LIBS = @FFTW3F_LIBS@
 
 FFTW_XCPPFLAGS = $(FFTW3_CPPFLAGS) $(FFTW3F_CPPFLAGS)
 FFTW_XLDFLAGS = $(FFTW3_LDFLAGS) $(FFTW3F_LDFLAGS)
 FFTW_XLIBS = $(FFTW3_LIBS) $(FFTW3F_LIBS)
 
-FLTK_CFLAGS = @FLTK_CFLAGS@
+FLTK_CPPFLAGS = @FLTK_CPPFLAGS@
+FLTK_LDFLAGS = @FLTK_LDFLAGS@
 FLTK_LIBS = @FLTK_LIBS@
 
 ## Alias CPPFLAGS to CFLAGS.
 ## This is closer to the true meaning of `pkg-config --cflags` output.
 FONTCONFIG_CPPFLAGS = @FONTCONFIG_CFLAGS@
 FONTCONFIG_CFLAGS = @FONTCONFIG_CFLAGS@
 
 ## Alias CPPFLAGS to CFLAGS.
@@ -551,21 +552,23 @@ echo "making $@ from $<"
   -e "s|%OCTAVE_CONF_FFLAGS%|\"${FFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_FFTW3_CPPFLAGS%|\"${FFTW3_CPPFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_FFTW3_LDFLAGS%|\"${FFTW3_LDFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_FFTW3_LIBS%|\"${FFTW3_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_FFTW3F_CPPFLAGS%|\"${FFTW3F_CPPFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_FFTW3F_LDFLAGS%|\"${FFTW3F_LDFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_FFTW3F_LIBS%|\"${FFTW3F_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_FLIBS%|\"${FLIBS}\"|" \
-  -e "s|%OCTAVE_CONF_FLTK_CFLAGS%|\"${FLTK_CFLAGS}\"|" \
+  -e "s|%OCTAVE_CONF_FLTK_CPPFLAGS%|\"${FLTK_CPPFLAGS}\"|" \
+  -e "s|%OCTAVE_CONF_FLTK_LDFLAGS%|\"${FLTK_LDFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_FLTK_LIBS%|\"${FLTK_LIBS}\"|" \
-  -e "s|%OCTAVE_CONF_FONTCONFIG_CFLAGS%|\"${FONTCONFIG_CFLAGS}\"|" \
+  -e "s|%OCTAVE_CONF_FONTCONFIG_CPPFLAGS%|\"${FONTCONFIG_CPPFLAGS}\"|" \
+  -e "s|%OCTAVE_CONF_FONTCONFIG_LIBS%|\"${FONTCONFIG_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_FPICFLAG%|\"${FPICFLAG}\"|" \
-  -e "s|%OCTAVE_CONF_FT2_CFLAGS%|\"${FT2_CFLAGS}\"|" \
+  -e "s|%OCTAVE_CONF_FT2_CPPFLAGS%|\"${FT2_CPPFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_FT2_LIBS%|\"${FT2_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_GLPK_CPPFLAGS%|\"${GLPK_CPPFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_GLPK_LDFLAGS%|\"${GLPK_LDFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_GLPK_LIBS%|\"${GLPK_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_GNUPLOT%|\"${GNUPLOT}\"|" \
   -e "s|%OCTAVE_CONF_HDF5_CPPFLAGS%|\"${HDF5_CPPFLAGS}\"|" | \
   $(SED) -e "s|%OCTAVE_CONF_HDF5_LDFLAGS%|\"${HDF5_LDFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_HDF5_LIBS%|\"${HDF5_LIBS}\"|" \
diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -1167,18 +1167,19 @@ AC_ARG_WITH([opengl],
    fi],
   [check_opengl=yes])
 
 ## Check for OpenGL library
 if test $check_opengl = yes; then
   OCTAVE_CHECK_LIB_OPENGL
 fi
 
+FLTK_CPPFLAGS=
+FLTK_LDFLAGS=
 FLTK_LIBS=
-FLTK_CFLAGS=
 
 if test -z "$OPENGL_LIBS"; then
   if test $check_opengl = yes; then
     native_graphics=no
     warn_fltk_opengl="OpenGL libs (GL and GLU) not found.  Native graphics will be disabled."
     OCTAVE_CONFIGURE_WARNING([warn_fltk_opengl])
   fi
 fi
@@ -1203,16 +1204,18 @@ if test -n "$OPENGL_LIBS"; then
     else
       AC_MSG_RESULT(no)
       warn_freetype="FreeType library >= 9.03 not found.  Native graphics will be disabled."
     fi])
 
   if test -n "$warn_freetype"; then
     native_graphics=no
     OCTAVE_CONFIGURE_WARNING([warn_freetype])
+  else
+    FT2_CPPFLAGS="$FT2_CFLAGS"
   fi
 
   ## Check for fontconfig library
 
   warn_fontconfig=""
   if test $native_graphics = yes; then
     PKG_CHECK_MODULES(FONTCONFIG, [fontconfig],
       [have_fontconfig=yes
@@ -1220,16 +1223,18 @@ if test -n "$OPENGL_LIBS"; then
        AC_DEFINE(HAVE_FONTCONFIG, 1, [Define to 1 if fontconfig is present.])],
       [have_fontconfig=no
        warn_fontconfig="Fontconfig library not found.  Native graphics will be disabled."])
   fi
 
   if test -n "$warn_fontconfig"; then
     native_graphics=no
     OCTAVE_CONFIGURE_WARNING([warn_fontconfig])
+  else
+    FONTCONFIG_CPPFLAGS="$FONTCONFIG_CFLAGS"
   fi
 
   ## Check for Xft library (when using X11)
 
   warn_xft=""
   if test $native_graphics = yes && test "$have_x" = yes; then
     PKG_CHECK_MODULES(XFT, [xft],
       [AC_DEFINE(HAVE_XFT, 1, [Define to 1 if Xft is present.])],
@@ -1288,53 +1293,54 @@ if test -n "$OPENGL_LIBS"; then
     warn_fltk_config=""
     warn_fltk_opengl=""
 
     if test "$FLTK_CONFIG" = no; then
       native_graphics=no
       warn_fltk_config="FLTK config script not found.  Native graphics will be disabled."
       OCTAVE_CONFIGURE_WARNING([warn_fltk_config])
     else
-      FLTK_CFLAGS=`$FLTK_CONFIG $fltkconf_args --use-gl --cflags`
+      FLTK_CPPFLAGS=`$FLTK_CONFIG $fltkconf_args --use-gl --cflags`
       FLTK_LDFLAGS=`$FLTK_CONFIG $fltkconf_args --use-gl --ldflags`
 
       case $host_os in
         mingw*)
           FLTK_LDFLAGS=`echo $FLTK_LDFLAGS | $SED -e 's/-mwindows//g'`
         ;;
       esac
 
       AC_CACHE_CHECK([for OpenGL support in FLTK],
         [octave_cv_fltk_opengl_support],
-        [save_CFLAGS="$CFLAGS"
-        CFLAGS="$CFLAGS $FLTK_CFLAGS"
+        [save_CPPFLAGS="$CPPFLAGS"
+        CPPFLAGS="$CFLAGS $FLTK_CPPFLAGS"
         AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
           #include <FL/gl.h>
           ]], [[
           int nothing = 0;
           ]])],
           octave_cv_fltk_opengl_support=no,
           octave_cv_fltk_opengl_support=yes)
-        CFLAGS="$save_CFLAGS"
+        CPPFLAGS="$save_CPPFLAGS"
         ])
       if test $octave_cv_fltk_opengl_support = no; then
         warn_fltk_opengl="FLTK does not have OpenGL support.  Native graphics will be disabled."
       else
         AC_DEFINE(HAVE_FLTK, 1, [Define to 1 if FLTK is available.])
       fi
 
       if test -n "$warn_fltk_opengl"; then
         native_graphics=no
         OCTAVE_CONFIGURE_WARNING([warn_fltk_opengl])
       fi
     fi
   fi
 fi
 
-AC_SUBST(FLTK_CFLAGS)
+AC_SUBST(FLTK_CPPFLAGS)
+AC_SUBST(FLTK_LDFLAGS)
 AC_SUBST(FLTK_LIBS)
 
 ## Check for gl2ps which is required for printing with OpenGL graphics
 if test $native_graphics = yes; then
   AC_CHECK_HEADERS([gl2ps.h],
     [GL2PS_LIBS="-lgl2ps"],
     [warn_gl2ps="gl2ps library not found.  OpenGL printing is disabled."
      OCTAVE_CONFIGURE_WARNING([warn_gl2ps])])
@@ -3019,21 +3025,22 @@ Octave is now configured for $canonical_
   CXSPARSE libraries:          $CXSPARSE_LIBS
   DL libraries:                $DL_LIBS
   FFTW3 CPPFLAGS:              $FFTW3_CPPFLAGS
   FFTW3 LDFLAGS:               $FFTW3_LDFLAGS
   FFTW3 libraries:             $FFTW3_LIBS
   FFTW3F CPPFLAGS:             $FFTW3F_CPPFLAGS
   FFTW3F LDFLAGS:              $FFTW3F_LDFLAGS
   FFTW3F libraries:            $FFTW3F_LIBS
-  FLTK CFLAGS:                 $FLTK_CFLAGS
+  FLTK CPPFLAGS:               $FLTK_CPPFLAGS
+  FLTK LDFLAGS:                $FLTK_LDFLAGS
   FLTK libraries:              $FLTK_LIBS
-  fontconfig CFLAGS:           $FONTCONFIG_CFLAGS
+  fontconfig CPPFLAGS:         $FONTCONFIG_CPPFLAGS
   fontconfig libraries:        $FONTCONFIG_LIBS
-  FreeType2 CFLAGS:            $FT2_CFLAGS
+  FreeType2 CPPFLAGS:          $FT2_CPPFLAGS
   FreeType2 libraries:         $FT2_LIBS
   GLPK CPPFLAGS:               $GLPK_CPPFLAGS
   GLPK LDFLAGS:                $GLPK_LDFLAGS
   GLPK libraries:              $GLPK_LIBS
   HDF5 CPPFLAGS:               $HDF5_CPPFLAGS
   HDF5 LDFLAGS:                $HDF5_LDFLAGS
   HDF5 libraries:              $HDF5_LIBS
   Java home:                   $JAVA_HOME
diff --git a/libinterp/corefcn/toplev.cc b/libinterp/corefcn/toplev.cc
--- a/libinterp/corefcn/toplev.cc
+++ b/libinterp/corefcn/toplev.cc
@@ -1339,21 +1339,23 @@ specified option.\n\
     { false, "FFLAGS", OCTAVE_CONF_FFLAGS },
     { false, "FFTW3_CPPFLAGS", OCTAVE_CONF_FFTW3_CPPFLAGS },
     { false, "FFTW3_LDFLAGS", OCTAVE_CONF_FFTW3_LDFLAGS },
     { false, "FFTW3_LIBS", OCTAVE_CONF_FFTW3_LIBS },
     { false, "FFTW3F_CPPFLAGS", OCTAVE_CONF_FFTW3F_CPPFLAGS },
     { false, "FFTW3F_LDFLAGS", OCTAVE_CONF_FFTW3F_LDFLAGS },
     { false, "FFTW3F_LIBS", OCTAVE_CONF_FFTW3F_LIBS },
     { false, "FLIBS", OCTAVE_CONF_FLIBS },
-    { false, "FLTK_CFLAGS", OCTAVE_CONF_FLTK_CFLAGS },
+    { false, "FLTK_CPPFLAGS", OCTAVE_CONF_FLTK_CPPFLAGS },
+    { false, "FLTK_LDFLAGS", OCTAVE_CONF_FLTK_LDFLAGS },
     { false, "FLTK_LIBS", OCTAVE_CONF_FLTK_LIBS },
-    { false, "FONTCONFIG_CFLAGS", OCTAVE_CONF_FONTCONFIG_CFLAGS },
+    { false, "FONTCONFIG_CPPFLAGS", OCTAVE_CONF_FONTCONFIG_CPPFLAGS },
+    { false, "FONTCONFIG_LIBS", OCTAVE_CONF_FONTCONFIG_LIBS },
     { false, "FPICFLAG", OCTAVE_CONF_FPICFLAG },
-    { false, "FT2_CFLAGS", OCTAVE_CONF_FT2_CFLAGS },
+    { false, "FT2_CPPFLAGS", OCTAVE_CONF_FT2_CPPFLAGS },
     { false, "FT2_LIBS", OCTAVE_CONF_FT2_LIBS },
     { false, "GLPK_CPPFLAGS", OCTAVE_CONF_GLPK_CPPFLAGS },
     { false, "GLPK_LDFLAGS", OCTAVE_CONF_GLPK_LDFLAGS },
     { false, "GLPK_LIBS", OCTAVE_CONF_GLPK_LIBS },
     { false, "GNUPLOT", OCTAVE_CONF_GNUPLOT },
     { false, "HDF5_CPPFLAGS", OCTAVE_CONF_HDF5_CPPFLAGS },
     { false, "HDF5_LDFLAGS", OCTAVE_CONF_HDF5_LDFLAGS },
     { false, "HDF5_LIBS", OCTAVE_CONF_HDF5_LIBS },
diff --git a/libinterp/dldfcn/module-files b/libinterp/dldfcn/module-files
--- a/libinterp/dldfcn/module-files
+++ b/libinterp/dldfcn/module-files
@@ -1,17 +1,17 @@
 # FILE|CPPFLAGS|LDFLAGS|LIBRARIES
 __delaunayn__.cc|$(QHULL_CPPFLAGS)|$(QHULL_LDFLAGS)|$(QHULL_LIBS)
 __dsearchn__.cc
 __eigs__.cc|$(ARPACK_CPPFLAGS) $(SPARSE_XCPPFLAGS)|$(ARPACK_LDFLAGS) $(SPARSE_XLDFLAGS)|$(ARPACK_LIBS) $(SPARSE_XLIBS) $(LAPACK_LIBS) $(BLAS_LIBS)
-__fltk_uigetfile__.cc|$(FLTK_CFLAGS) $(FT2_CPPFLAGS)|$(FLTK_LDFLAGS) $(FT2_LDFLAGS)|$(FLTK_LIBS) $(FT2_LIBS)
+__fltk_uigetfile__.cc|$(FLTK_CPPFLAGS) $(FT2_CPPFLAGS)|$(FLTK_LDFLAGS) $(FT2_LDFLAGS)|$(FLTK_LIBS) $(FT2_LIBS)
 __glpk__.cc|$(GLPK_CPPFLAGS)|$(GLPK_LDFLAGS)|$(GLPK_LIBS)
 __ichol__.cc
 __ilu__.cc
-__init_fltk__.cc|$(FLTK_CFLAGS) $(FT2_CPPFLAGS) $(FONTCONFIG_CPPFLAGS)|$(FLTK_LDFLAGS) $(FT2_LDFLAGS)|$(FLTK_LIBS) $(FT2_LIBS) $(OPENGL_LIBS)
+__init_fltk__.cc|$(FLTK_CPPFLAGS) $(FT2_CPPFLAGS) $(FONTCONFIG_CPPFLAGS)|$(FLTK_LDFLAGS) $(FT2_LDFLAGS)|$(FLTK_LIBS) $(FT2_LIBS) $(OPENGL_LIBS)
 __init_gnuplot__.cc|$(FT2_CPPFLAGS) $(FONTCONFIG_CPPFLAGS)||
 __magick_read__.cc|$(MAGICK_CPPFLAGS)|$(MAGICK_LDFLAGS)|$(MAGICK_LIBS)
 __osmesa_print__.cc|$(OSMESA_CPPFLAGS) $(FT2_CPPFLAGS)|$(OSMESA_LDFLAGS)|$(OSMESA_LIBS)
 __voronoi__.cc|$(QHULL_CPPFLAGS)|$(QHULL_LDFLAGS)|$(QHULL_LIBS)
 amd.cc|$(SPARSE_XCPPFLAGS)|$(SPARSE_XLDFLAGS)|$(SPARSE_XLIBS)
 ccolamd.cc|$(SPARSE_XCPPFLAGS)|$(SPARSE_XLDFLAGS)|$(SPARSE_XLIBS)
 chol.cc|$(SPARSE_XCPPFLAGS)|$(SPARSE_XLDFLAGS)|$(SPARSE_XLIBS)
 colamd.cc|$(SPARSE_XCPPFLAGS)|$(SPARSE_XLDFLAGS)|$(SPARSE_XLIBS)
diff --git a/libinterp/oct-conf.in.h b/libinterp/oct-conf.in.h
--- a/libinterp/oct-conf.in.h
+++ b/libinterp/oct-conf.in.h
@@ -273,34 +273,42 @@ along with Octave; see the file COPYING.
 #ifndef OCTAVE_CONF_FFTW3F_LIBS
 #define OCTAVE_CONF_FFTW3F_LIBS %OCTAVE_CONF_FFTW3F_LIBS%
 #endif
 
 #ifndef OCTAVE_CONF_FLIBS
 #define OCTAVE_CONF_FLIBS %OCTAVE_CONF_FLIBS%
 #endif
 
-#ifndef OCTAVE_CONF_FLTK_CFLAGS
-#define OCTAVE_CONF_FLTK_CFLAGS %OCTAVE_CONF_FLTK_CFLAGS%
+#ifndef OCTAVE_CONF_FLTK_CPPFLAGS
+#define OCTAVE_CONF_FLTK_CPPFLAGS %OCTAVE_CONF_FLTK_CPPFLAGS%
+#endif
+
+#ifndef OCTAVE_CONF_FLTK_LDFLAGS
+#define OCTAVE_CONF_FLTK_LDFLAGS %OCTAVE_CONF_FLTK_LDFLAGS%
 #endif
 
 #ifndef OCTAVE_CONF_FLTK_LIBS
 #define OCTAVE_CONF_FLTK_LIBS %OCTAVE_CONF_FLTK_LIBS%
 #endif
 
-#ifndef OCTAVE_CONF_FONTCONFIG_CFLAGS
-#define OCTAVE_CONF_FONTCONFIG_CFLAGS %OCTAVE_CONF_FONTCONFIG_CFLAGS%
+#ifndef OCTAVE_CONF_FONTCONFIG_CPPFLAGS
+#define OCTAVE_CONF_FONTCONFIG_CPPFLAGS %OCTAVE_CONF_FONTCONFIG_CPPFLAGS%
+#endif
+
+#ifndef OCTAVE_CONF_FONTCONFIG_LIBS
+#define OCTAVE_CONF_FONTCONFIG_LIBS %OCTAVE_CONF_FONTCONFIG_LIBS%
 #endif
 
 #ifndef OCTAVE_CONF_FPICFLAG
 #define OCTAVE_CONF_FPICFLAG %OCTAVE_CONF_FPICFLAG%
 #endif
 
-#ifndef OCTAVE_CONF_FT2_CFLAGS
-#define OCTAVE_CONF_FT2_CFLAGS %OCTAVE_CONF_FT2_CFLAGS%
+#ifndef OCTAVE_CONF_FT2_CPPFLAGS
+#define OCTAVE_CONF_FT2_CPPFLAGS %OCTAVE_CONF_FT2_CPPFLAGS%
 #endif
 
 #ifndef OCTAVE_CONF_FT2_LIBS
 #define OCTAVE_CONF_FT2_LIBS %OCTAVE_CONF_FT2_LIBS%
 #endif
 
 #ifndef OCTAVE_CONF_GLPK_CPPFLAGS
 #define OCTAVE_CONF_GLPK_CPPFLAGS %OCTAVE_CONF_GLPK_CPPFLAGS%
